state/protocol/blocktimer/blocktimer_test.go

```diff
@@ -0,0 +1,66 @@
+package blocktimer
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/onflow/flow-go/state/protocol"
+)
+
+// TestBlockTimestamp_Build tests that builder correctly generates new block time
+func TestBlockTimestamp_Build(t *testing.T) {
```

Comment:
Would suggest to test the Validation method first. Why? because we can first verify the Validation method is implemented correctly, then use it to verify the Build method, so that we could verify a created block time can pass the our own validation.

I would suggest to add more tests on boundary:

1. `now = parent - 1`
2. `now = parent`
1. `now = parent + 1`
3. `now = parent + minInterval - 1`
4. `now = parent + minInterval`
3. `now = parent + minInterval + 1`
4. `now = parent + maxInterval - 1`
5. `now = parent + maxInterval`
5. `now = parent + maxInterval + 1`

For all above cases, we should test that all of the created time should pass the `Validation` method call.




Why?

Because we can first verify the Validation method is implemented correctly, then use it to verify the Build method, so that we could verify a created block time can pass the our own validation.

The `Build` function internally calls `Now()` which makes it hard to test the boundary. If we can fix the `Now()`, then we can fix the boundary to be tested.